### PR TITLE
Switch to dojo

### LINKED ISSUE #6: Smart contracts design
# Description

According to the [game rules](https://github.com/keep-starknet-strange/tsubasa/discussions/2#discussioncomment-6567694) we need to design the smart contracts and the interface that they will define. This will allow the frontend integration and will set the base for parallelising the development.

# Acceptance criteria

* The smart contract is an empty shell that defines the right interface

### LINKED ISSUE #7: Create the smart-contract unit testing files
# Description

We need to unit test every function from the smart contracts. We'll first test the mocked entrypoints.

# Acceptance criteria

* Test files are created and cal the mocked entrypoints

### PR DESCRIPTION
**Pull request by LucasLvy**
Change the architecture of the cairo contracts to use the dojo framework
Fixes #7
Fixes #6

### CODE REVIEW

File: contracts/src/game.cairo [hunk 1]
```diff
@@ -1,44 +1,73 @@
...
+#[derive(Component, Copy, Drop, Serde, SerdeLen)]
+struct Game {
+    /// Rounds won by the player 1.
+    player1_score: u8,
+    /// Rounds won by the player 2.
+    player2_score: u8,
+    /// Current turn of the round.
+    turn: felt252,
+    /// Winner of the game. As long as it is `None` it means that the game is playing.
+    winner: Option<Player>,
```
Comment (tarrencev) [inline contracts/src/game.cairo:11]: Perhaps this should be a contract address?

File: contracts/src/game.cairo [hunk 2]
```diff
@@ -1,44 +1,73 @@
...
-    Player2: ContractAddress,
+    Player1: (),
+    Player2: (),
+    Draw: (),
+}
+
+impl PlayerSerdeLen of dojo::SerdeLen<Option<Player>> {
+    #[inline(always)]
+    fn len() -> usize {
+        0
```
Comment (tarrencev) [inline contracts/src/game.cairo:24]:
```suggestion
        1
```

This serializes to a single felt right?

File: contracts/src/card.cairo [hunk 3]
```diff
@@ -0,0 +1,55 @@
+/// Represents a playing card. It only contains the token id of the NFT.
+#[derive(Component, Copy, Drop, Serde, SerdeLen)]
+struct Card {
```
Comment (tarrencev) [inline contracts/src/card.cairo:3]: Hmm I think we can key this by the token id. We probably want it to contain player stats?

Comment (LucasLvy) [reply contracts/src/card.cairo:3]: idk if the player stats should be at this level or at the nft contract level

**Review by tarrencev — APPROVED** [review]

**Pull request merged**